lib/models/os.coffee

```diff
@@ -49,10 +52,13 @@ getOsModel = (deps, opts) ->
 			"/image/#{deviceType}/versions"
 		postProcess: ({ body }) ->
 			{ versions, latest } = body
-			versions.sort(semver.rcompare)
-			recommended = reject(versions, semver.prerelease)?[0] || null
+			[ validVersions, invalidVersions ] = partition(versions, semver.valid)
+
+			validVersions.sort(semver.rcompare)
+			recommended = reject(validVersions, semver.prerelease)?[0] || null
```

Comment:
@LucianBuzzo Looking at this makes me realise that whichever way we handle the version numbers from now on, we should be commonizing the OS version recommendation logic.